### ISSUE TITLE
More explicit warning in git sync

### ISSUE THREAD

**Issue by arl** [tj/git-extras#730]
The warning printed by `git sync` says:

```
Are you sure you want to clean all changes & sync with 'origin/master'? [y/N]:
```

Typing `Y` I was **not** expecting untracked files to be definitely removed. It may be due to the fact that English is not my mother tongue. This is what I was expecting `git sync` to do after reading both the man page and the warning:
 
- `fetch` the remote branch tracked by my local one, for instance `master` and `origin/master`
- `reset --hard` to to sync local `HEAD` with remote `HEAD`
- **not** removing any **untracked** files/directories

In _non-git words_ I was expecting only my tracked files to be sync'ed.

**Comment by sleagon** [comment]
@arl add a new option U for git sync.

**Comment by spacewander** [comment]
Solved in #739.
